### ISSUE TITLE
Deprecation warning

### ISSUE THREAD

**Issue by TTLucian** [ScratMan/HASmartThermostat#219]
**Describe the bug**
Deprecation warning in HA Llogs

```
WARNING (MainThread) [homeassistant.helpers.frame] Detected that custom integration 'smart_thermostat' calls `async_track_state_change` instead of `async_track_state_change_event` which is deprecated and will be removed in Home Assistant 2025.5 at custom_components/smart_thermostat/climate.py, line 380: async_track_state_change(, please create a bug report at https://github.com/ScratMan/HASmartThermostat/issues

```

**Comment by stanol** [comment]
I have two more warnings
```
Detected that custom integration 'smart_thermostat' calls `async_track_state_change` instead of `async_track_state_change_event` which is deprecated and will be removed in Home Assistant 2025.5 at custom_components/smart_thermostat/climate.py, line 369: async_track_state_change(, please create a bug report at https://github.com/ScratMan/HASmartThermostat/issues
Detected that custom integration 'smart_thermostat' calls `async_track_state_change` instead of `async_track_state_change_event` which is deprecated and will be removed in Home Assistant 2025.5 at custom_components/smart_thermostat/climate.py, line 375: async_track_state_change(, please create a bug report at https://github.com/ScratMan/HASmartThermostat/issues
```

**Comment by ScratMan** [comment]
Fixed in 2024.6.0
